samples/bluetooth/nrf_dm/src/peer.c

```diff
@@ -74,10 +76,24 @@ static struct peer_entry *mcpd_min_peer_result(struct peer_entry *a, struct peer
 		return a;
 	}
 
+#ifdef CONFIG_DM_HIGH_PRECISION_CALC
+	if (!isnan(a->result.dist_estimates.mcpd.high_precision) ||
+	    !isnan(b->result.dist_estimates.mcpd.high_precision)) {
+
+		if (a->result.dist_estimates.mcpd.high_precision >
```

Comment:
If a or b is NaN then it will execute:
```
	if (a->result.dist_estimates.mcpd.best < b->result.dist_estimates.mcpd.best) {
		return a;
	} else {
		return b;
	}
```
Corrected the condition from `or` to `and`.

Reply:
I think we would still be able to use the high_precision estimate of the other if only one of them is NaN.